x-pack/plugins/enterprise_search/public/applications/app_search/components/documents/search_experience/customization_modal.tsx

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useState } from 'react';
+
+import {
+  EuiButton,
+  EuiButtonEmpty,
+  EuiComboBox,
+  EuiForm,
+  EuiFormRow,
+  EuiModal,
+  EuiModalBody,
+  EuiModalFooter,
+  EuiModalHeader,
+  EuiModalHeaderTitle,
+  EuiOverlayMask,
+  EuiText,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { useValues } from 'kea';
+
+import { EngineLogic } from '../../engine';
+
+interface Props {
+  filterFields: string[];
+  sortFields: string[];
+  onClose(): void;
+  onSave({ sortFields, filterFields }: { sortFields: string[]; filterFields: string[] }): void;
+}
+
+const fieldNameToComboBoxOption = (fieldName: string) => ({ label: fieldName });
+const comboBoxOptionToFieldName = ({ label }: { label: string }) => label;
+
+export const CustomizationModal: React.FC<Props> = ({
+  filterFields,
+  onClose,
+  onSave,
+  sortFields,
+}) => {
+  const { engine } = useValues(EngineLogic);
+
+  const [selectedFilterFields, setSelectedFilterFields] = useState(
+    filterFields.map(fieldNameToComboBoxOption)
+  );
+  const [selectedSortFields, setSelectedSortFields] = useState(
+    sortFields.map(fieldNameToComboBoxOption)
+  );
+
+  const engineSchema = engine.schema || {};
+  const selectableFilterFields = Object.keys(engineSchema).map(fieldNameToComboBoxOption);
+  const selectableSortFields = Object.keys(engineSchema).map(fieldNameToComboBoxOption);
```

Comment:
This is a potentially expensive operation, let's `useMemo` it since I predict that engine.schema will be changing far less often than the current selected filters/fields

```suggestion

  const selectableFilterFields = useMemo(
    () => Object.keys(engineSchema).map(fieldNameToComboBoxOption),
    [engineSchema]
  );
  const selectableSortFields = useMemo(
    () => Object.keys(engineSchema).map(fieldNameToComboBoxOption),
    [engineSchema]
  );
```

Reply:
c54afb6f6e5a9b5289053f40e5830ffa9f8c3bb8